src/Polly.Core/Simmy/Outcomes/OutcomeChaosStrategy.cs

```diff
@@ -0,0 +1,132 @@
+                    if (fault is not null)
+                    {
+                        return new Outcome<T>(fault);
+                    }
+                }
+                else if (OutcomeGenerator is not null)
+                {
+                    var outcome = await InjectOutcome(context).ConfigureAwait(context.ContinueOnCapturedContext);
+                    if (outcome.HasValue)
+                    {
+                        return new Outcome<T>(outcome.Value.Result);
+                    }
+                }
+                else
+                {
+                    throw new InvalidOperationException("Either a fault or fake outcome to inject must be defined.");
+                }
+            }
+
+            return await StrategyHelper.ExecuteCallbackSafeAsync(callback, context, state).ConfigureAwait(context.ContinueOnCapturedContext);
+        }
+        catch (OperationCanceledException e)
+        {
+            return new Outcome<T>(e);
+        }
+    }
+
+    private async ValueTask<Outcome<T>?> InjectOutcome(ResilienceContext context)
+    {
+        var outcome = await OutcomeGenerator!(new(context)).ConfigureAwait(context.ContinueOnCapturedContext);
+        var args = new OutcomeArguments<T, OnOutcomeInjectedArguments>(context, outcome.Value, new(context));
+        _telemetry.Report(new(ResilienceEventSeverity.Warning, OutcomeConstants.OnOutcomeInjectedEvent), context, args);
+
+        if (OnOutcomeInjected is not null)
+        {
+            await OnOutcomeInjected(args).ConfigureAwait(context.ContinueOnCapturedContext);
+        }
+
+        return outcome;
+    }
+
+    private async ValueTask<Exception?> InjectFault(ResilienceContext context)
+    {
+        try
+        {
+            var fault = await FaultGenerator!(new(context)).ConfigureAwait(context.ContinueOnCapturedContext);
+            if (!fault.HasValue)
+            {
+                return null;
+            }
+
+            // to prevent injecting the fault if it was cancelled while executing the FaultGenerator
+            context.CancellationToken.ThrowIfCancellationRequested();
+
+            Outcome = new(fault.Value.Exception!);
+            var args = new OutcomeArguments<Exception, OnOutcomeInjectedArguments>(context, new Outcome<Exception>(fault.Value.Exception!), new(context));
+            _telemetry.Report(new(ResilienceEventSeverity.Warning, OutcomeConstants.OnFaultInjectedEvent), context, args);
+
+            if (OnFaultInjected is not null)
+            {
+                await OnFaultInjected(args).ConfigureAwait(context.ContinueOnCapturedContext);
+            }
+
+            return fault.Value.Exception;
+        }
+        catch (OperationCanceledException)
+        {
+            // fault injection might be cancelled during FaultGenerator, if so we run the user's delegate normally
+            context.CancellationToken = CancellationToken.None;
```

Comment:
Can we drop this change? As of now the timeout strategy is th eonly one that does changes to cancellation token. I feel this shouldn't belong here.